rfcs/055-prismic-api/README.md

```diff
@@ -0,0 +1,249 @@
+- The elasticsearch index mapping should represent the contract between the pipeline and the API. The API shouldn't need to know anything about the structure of the data in Prismic.
+- We won't include an endpoint for `/series`, nor will we include series nodes in the `/articles` endpoint. If an article is part of a series, we'll denormalise information from its parent onto the article itself to use for relevance and visual signposting in results.
+
+## Proposed endpoints
+
+### `/articles`
+
+#### Example response
+
+```http
+GET /articles/{id}
+{
+    "id": "",
+    "format": "",
+    "title": "",
+    "body": "", // not exposed by default, should require an ?includes=body
+    "standfirst": "",
+    "published": "",
+    "contributors": [
+        {
+            ... // reuse the Contributor model
+        },
+        ...
+    ],
+    "image": {
+        "thumbnail": {
+            ... // reuse the DigitalLocation model
+        },
+        "caption": ""
+    },
+    "partOf": [
+        {
+            ... // reuse the partOf model for series
+        }
+    ]
+}
+```
+
+```http
+GET /articles?query=foo&filters=bar&sort=baz
+{
+    "type": "ResultList",
+    "results": [
+        {
+            ... // as above
+        },
+        ...
+    ],
+    "pageSize": 100,
+    "totalResults": 1000,
+    "totalPages": 10,
+    "nextPage": "/articles?query=foo&filters=bar&sort=baz?page=2",
+}
+```
+
+#### Filters
+
+- date
+- contributor
+- series
+- contentType
+
+#### Sort
+
+- date
+- relevance
+
+### `/exhibitions`
+
+#### Example response
+
+```http
+GET /exhibitions/{id}
+{
+    "id": "",
+    "format": "",
+    "title": "",
+    "status": "",
+    "instantiations": [
+        {
+            "start": "", // ISO 8601 date only (no time)
+            "end": "", // ISO 8601 date only (no time)
+            "place": "",
+            "guide": {
+                ... // reuse the DigitalLocation model
+            }
+        }
+    ],
+    "contributors": [ // behind ?includes=contributors
+        {
+            ... // reuse the Contributor model
+        },
+        ...
+    ],
+    "image": {
+        "thumbnail": {
+            ... // reuse the DigitalLocation model
+        },
+        "caption": ""
+    },
+    "seasons": [
+        "",
+        ...
+    ],
+}
+```
+
+```http
+GET /exhibitions?query=foo&filters=bar&sort=baz
+{
+    "type": "ResultList",
+    "results": [
+        {
+            ... // as above
+        },
+        ...
+    ],
+    "pageSize": 100,
+    "totalResults": 1000,
+    "totalPages": 10,
+    "nextPage": "/exhibitions?query=foo&filters=bar&sort=baz?page=2",
+}
+```
+
+#### Filter
+
+- date
+- status
+- place
+- contributor
+- format (ie permanent, exhibition, installation)
+- season
+
+#### Sort
+
+- date
+- relevance
+
+### `/events`
+
+#### Example response
+
+```http
+GET /events/{id}
+{
+    "id": "",
+    "format": "",
+    "title": "",
+    "status": "",
+    "isOnline": false,
+    "availableOnline": false,
```

Comment:
What do these mean?

Reply:
From what I could gather; `isOnline` means the event takes place online
`availableOnline` means it's available for viewing afterwards (was recorded).

cc @gestchild @davidpmccormick for confirmation?

Reply:
What are they in Prismic? I'm assuming it helps to render links to events without having to look at all that, much like the availabilities on a work? I can also see us wanting to filter events in a simple way like this.

Reply:
Another example of why we need to see some real examples here.

Reply:
**General comment about events:**

I think the way events are modelled in Prismic is somewhat sub-optimal and not something we should necessarily seek to emulate; I know Jenn is thinking about this also.

see e.g. https://github.com/wellcomecollection/wellcomecollection.org/issues/8902 and https://github.com/wellcomecollection/wellcomecollection.org/issues/8830 for some thorny examples

I agree with sticking to Prismic in general, but this might be a case where we want to make changes with a view to backporting those back into Prismic later.

Before you can model this in the API, I think we need to think about what behaviour we want out of the front-end. e.g. the "Lights Up" session in those tickets that runs on three different occasions but has a single summary page on the website. How does that show up when I use the website search? How does it interact with filters? Does the appearance of the search result change based on my filters? etc.